### PR TITLE
Add Kodi snap for stable release.

### PR DESCRIPTION
**Pull request by dgmvecuador**
Add a snap for Kodi for the last stable tag "16.1-Jarvis"
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-235649738%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-235650004%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-235834367%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-235835667%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72476818%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72477168%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72560202%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72560474%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72583865%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72584104%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72584731%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-235917496%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-235927572%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-236088739%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-236136550%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-236198485%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-236514288%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-236594998%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-237247804%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-238532848%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-238765087%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23issuecomment-235649738%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22In%20Travis%20this%20fails%20to%20build%20with%20%5Cr%5Cn%60%60%60%5Cr%5Cnchecking%20mpeg2dec/mpeg2.h%20usability...%20no%5Cr%5Cnchecking%20mpeg2dec/mpeg2.h%20presence...%20no%5Cr%5Cnchecking%20for%20mpeg2dec/mpeg2.h...%20no%5Cr%5Cnconfigure%3A%20error%3A%20Could%20not%20find%20a%20required%20library.%20Please%20see%20the%20README%20for%20your%20platform.%5Cr%5CnCommand%20%27%5B%27/bin/sh%27%2C%20%27/tmp/tmpudwgxd8_%27%2C%20%27./configure%27%2C%20%27--prefix%3D%27%2C%20%27--prefix%3D/%27%5D%27%20returned%20non-zero%20exit%20status%201%60%60%60%22%2C%20%22created_at%22%3A%20%222016-07-27T16%3A57%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22On%20my%20machine%20locally%20it%20fails%20with%5Cr%5Cn%60%60%60%5Cr%5Cnchecking%20for%20LIBDCADEC...%20no%5Cr%5Cnconfigure%3A%20error%3A%20libdcadec%20not%20found%5Cr%5CnCommand%20%27%5B%27/bin/sh%27%2C%20%27/tmp/tmpwwmf3jxr%27%2C%20%27./configure%27%2C%20%27--prefix%3D%27%2C%20%27--prefix%3D/%27%5D%27%20returned%20non-zero%20exit%20status%201%5Cr%5Cn%60%60%60%22%2C%20%22created_at%22%3A%20%222016-07-27T16%3A57%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Can%20we%20somehow%20adjust%20where%20ffmpeg%20comes%20from%3F%20This%20is%20how%20the%20build%20fails%20%28both%20using%20%27snapcraft%27%20or%20%27snapcraft%20cleanbuild%27%29%3A%20http%3A//paste.ubuntu.com/21252125/%22%2C%20%22created_at%22%3A%20%222016-07-28T08%3A37%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-28T08%3A44%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Kodi%20developers%20usually%20recommend%20to%20use%20version%20of%20ffmpeg%20provided%20inside%20the%20code%2C%20so%20we%20should%20use%20that%20one.%20%20They%20update%20ffmpeg%20each%20major%20release.%22%2C%20%22created_at%22%3A%20%222016-07-28T14%3A47%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/406504%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dgmvecuador%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ok%2C%20let%27s%20pick%20whatever%20works.%20Can%20you%20apply%20the%20patch%20I%20specified%20earlier%3F%20That%20way%20you%20should%20see%20in%20the%20next%20Travis%20build%20how%20it%27s%20currently%20failing.%20Currently%20it%27s%20missing%20some%20build-packages.%22%2C%20%22created_at%22%3A%20%222016-07-28T15%3A19%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20have%20applied%20your%20patch%20and%20other%20changes.%20Now%20the%20build%20is%20successful%20in%20Travis%22%2C%20%22created_at%22%3A%20%222016-07-29T03%3A54%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/406504%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dgmvecuador%22%7D%7D%2C%20%7B%22body%22%3A%20%22Great%20work%21%5Cr%5Cn%5Cr%5CnWhen%20I%20try%20to%20run%20it%2C%20I%20see%20this%20though%3A%5Cr%5Cndaniel%40daydream%3A%7E/dev/snappy/snappy-playpen/kodi-stable%24%20kodi%5Cr%5CnlibGL%20error%3A%20unable%20to%20load%20driver%3A%20i965_dri.so%5Cr%5CnlibGL%20error%3A%20driver%20pointer%20missing%5Cr%5CnlibGL%20error%3A%20failed%20to%20load%20driver%3A%20i965%5Cr%5CnlibGL%20error%3A%20unable%20to%20load%20driver%3A%20swrast_dri.so%5Cr%5CnlibGL%20error%3A%20failed%20to%20load%20driver%3A%20swrast%5Cr%5Cn%5Cr%5CnAnd%20I%20get%20a%20black%20screen.%20Anything%20I%27m%20missing%3F%22%2C%20%22created_at%22%3A%20%222016-07-29T09%3A31%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Its%20a%20problem%20with%20opengl%20libs%2C%20there%20are%20multiple%20bugs%20reported%20like%20https%3A//bugs.launchpad.net/snappy/%2Bbug/1574851%5Cr%5Cn%5Cr%5CnAs%20workaround%2C%20you%20can%20copy%20your%20driver%20libs%20to%20the%20folder%20%5C%22/var/lib/snapd/lib/gl/%5C%22%20in%20my%20case%2C%20I%20copied%20all%20the%20content%20of%20%5C%22/usr/lib/nvidia-xxx%5C%22.%20I%20dont%20know%20about%20your%20graphic%20card%2C%20but%20it%20seems%20to%20work%20with%20intel%20without%20problems.%22%2C%20%22created_at%22%3A%20%222016-07-29T14%3A40%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/406504%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dgmvecuador%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%27m%20on%20intel%20too%20%28i915%29.%22%2C%20%22created_at%22%3A%20%222016-08-01T08%3A01%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22You%20should%20copy%20all%20the%20libs%20from%20Intel%2C%20%5C%22i965_dri.so%5C%22%20and%20the%20dependencies%20to%20%5C%22/var/lib/snapd/lib/gl/%5C%22%2C%20maybe%20check%20the%20deb%20package%20from%20the%20repository.%22%2C%20%22created_at%22%3A%20%222016-08-01T14%3A23%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/406504%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dgmvecuador%22%7D%7D%2C%20%7B%22body%22%3A%20%22Hi%2C%20did%20you%20solve%20the%20problem%20in%20your%20environment%3F%3F%22%2C%20%22created_at%22%3A%20%222016-08-03T14%3A10%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/406504%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dgmvecuador%22%7D%7D%2C%20%7B%22body%22%3A%20%22Sorry%20for%20the%20long%20wait.%20This%20makes%20it%20work%20for%20me%3A%20http%3A//paste.ubuntu.com/22795846/%22%2C%20%22created_at%22%3A%20%222016-08-09T12%3A05%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Brilliant%20work%21%20%3A-D%22%2C%20%22created_at%22%3A%20%222016-08-10T04%3A48%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2043e19fe8398103a3994e25cc56e52622e9405ccb%20kodi-stable/snapcraft.yaml%2049%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72583865%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Can%20you%20apply%20this%3F%20http%3A//paste.ubuntu.com/21252024/%5Cr%5Cn%5Cr%5CnIt%20add%27s%20a%20few%20%60build-packages%60.%22%2C%20%22created_at%22%3A%20%222016-07-28T08%3A36%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20kodi-stable/snapcraft.yaml%3AL1-209%22%7D%2C%20%22Pull%20abebc4a09f34fd3c26b67a0859efc2342aaf69dc%20kodi-stable/README.md%2026%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72476818%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Do%20you%20know%20if%20they%20were%20discussed%20on%20snapcraft%40lists.snapcraft.io%3F%20It%20looks%20like%20we%20might%20need%20another%20interface%20or%20two.%22%2C%20%22created_at%22%3A%20%222016-07-27T16%3A48%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20dont%20think%20so%2C%20maybe%20is%20a%20new%20issue%20and%20I%20havent%20reported%20it%22%2C%20%22created_at%22%3A%20%222016-07-28T03%3A20%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/406504%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dgmvecuador%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20think%20it%27d%20be%20good%20to%20at%20least%20bring%20it%20up%20there%20and%20see%20if%20others%20can%20work%20on%20resolving%20them.%22%2C%20%22created_at%22%3A%20%222016-07-28T08%3A38%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20kodi-stable/README.md%3AL1-30%22%7D%2C%20%22Pull%20abebc4a09f34fd3c26b67a0859efc2342aaf69dc%20kodi-stable/setup/gui/kodi.desktop%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/197%23discussion_r72477168%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Do%20these%20actions%20work%20as%20expected%3F%22%2C%20%22created_at%22%3A%20%222016-07-27T16%3A50%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes%2C%20they%20work%20as%20expected%5Cr%5Cn%5Cr%5Cn%21%5Bkodi_desktop_snap%5D%28https%3A//cloud.githubusercontent.com/assets/406504/17200199/e016032c-5448-11e6-852f-acd7bef7fba1.png%29%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222016-07-28T03%3A25%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/406504%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dgmvecuador%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-07-28T08%3A44%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1346979%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dholbach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20kodi-stable/setup/gui/kodi.desktop%3AL1-23%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull abebc4a09f34fd3c26b67a0859efc2342aaf69dc kodi-stable/README.md 26'></a> &nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/197#discussion_r72476818'>File: kodi-stable/README.md:L1-30</a></b>
- <a href='https://github.com/dholbach'></a> Do you know if they were discussed on snapcraft@lists.snapcraft.io? It looks like we might need another interface or two.
- <a href='https://github.com/dgmvecuador'></a> I dont think so, maybe is a new issue and I havent reported it
- <a href='https://github.com/dholbach'></a> I think it'd be good to at least bring it up there and see if others can work on resolving them.
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/197#issuecomment-235649738'>General Comment</a></b>
- <a href='https://github.com/dholbach'></a> In Travis this fails to build with
```
checking mpeg2dec/mpeg2.h usability... no
checking mpeg2dec/mpeg2.h presence... no
checking for mpeg2dec/mpeg2.h... no
configure: error: Could not find a required library. Please see the README for your platform.
Command '['/bin/sh', '/tmp/tmpudwgxd8_', './configure', '--prefix=', '--prefix=/']' returned non-zero exit status 1```
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> On my machine locally it fails with
```
checking for LIBDCADEC... no
configure: error: libdcadec not found
Command '['/bin/sh', '/tmp/tmpwwmf3jxr', './configure', '--prefix=', '--prefix=/']' returned non-zero exit status 1
```
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> Can we somehow adjust where ffmpeg comes from? This is how the build fails (both using 'snapcraft' or 'snapcraft cleanbuild'): http://paste.ubuntu.com/21252125/
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> :+1:<a href='#crh-update-none'></a>
- <a href='https://github.com/dgmvecuador'><img border=0 src='https://avatars.githubusercontent.com/u/406504?v=3' height=16 width=16'></a> Kodi developers usually recommend to use version of ffmpeg provided inside the code, so we should use that one.  They update ffmpeg each major release.
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> Ok, let's pick whatever works. Can you apply the patch I specified earlier? That way you should see in the next Travis build how it's currently failing. Currently it's missing some build-packages.
- <a href='https://github.com/dgmvecuador'><img border=0 src='https://avatars.githubusercontent.com/u/406504?v=3' height=16 width=16'></a> I have applied your patch and other changes. Now the build is successful in Travis
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> Great work!
When I try to run it, I see this though:
daniel@daydream:~/dev/snappy/snappy-playpen/kodi-stable$ kodi
libGL error: unable to load driver: i965_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: i965
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
And I get a black screen. Anything I'm missing?
- <a href='https://github.com/dgmvecuador'><img border=0 src='https://avatars.githubusercontent.com/u/406504?v=3' height=16 width=16'></a> Its a problem with opengl libs, there are multiple bugs reported like https://bugs.launchpad.net/snappy/+bug/1574851
As workaround, you can copy your driver libs to the folder "/var/lib/snapd/lib/gl/" in my case, I copied all the content of "/usr/lib/nvidia-xxx". I dont know about your graphic card, but it seems to work with intel without problems.
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> I'm on intel too (i915).
- <a href='https://github.com/dgmvecuador'><img border=0 src='https://avatars.githubusercontent.com/u/406504?v=3' height=16 width=16'></a> You should copy all the libs from Intel, "i965_dri.so" and the dependencies to "/var/lib/snapd/lib/gl/", maybe check the deb package from the repository.
- <a href='https://github.com/dgmvecuador'><img border=0 src='https://avatars.githubusercontent.com/u/406504?v=3' height=16 width=16'></a> Hi, did you solve the problem in your environment??
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> Sorry for the long wait. This makes it work for me: http://paste.ubuntu.com/22795846/
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> Brilliant work! :-D
- [ ] <a href='#crh-comment-Pull 43e19fe8398103a3994e25cc56e52622e9405ccb kodi-stable/snapcraft.yaml 49'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/197#discussion_r72583865'>File: kodi-stable/snapcraft.yaml:L1-209</a></b>
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> Can you apply this? http://paste.ubuntu.com/21252024/
It add's a few `build-packages`.
- [x] <a href='#crh-comment-Pull abebc4a09f34fd3c26b67a0859efc2342aaf69dc kodi-stable/setup/gui/kodi.desktop 14'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/197#discussion_r72477168'>File: kodi-stable/setup/gui/kodi.desktop:L1-23</a></b>
- <a href='https://github.com/dholbach'><img border=0 src='https://avatars.githubusercontent.com/u/1346979?v=3' height=16 width=16'></a> Do these actions work as expected?
- <a href='https://github.com/dgmvecuador'><img border=0 src='https://avatars.githubusercontent.com/u/406504?v=3' height=16 width=16'></a> Yes, they work as expected
![kodi_desktop_snap](https://cloud.githubusercontent.com/assets/406504/17200199/e016032c-5448-11e6-852f-acd7bef7fba1.png)


<a href='https://www.codereviewhub.com/ubuntu/snappy-playpen/pull/197?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/ubuntu/snappy-playpen/pull/197?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/ubuntu/snappy-playpen/pull/197'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by didrocks** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ubuntu/snappy-playpen/pull/197'>Refresh</a>.<a href='#crh-update-none'></a>

File: kodi-stable/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+Jul 18 21:35:49 XXXXXXXX kernel: [ 5027.855315]  exe="/usr/bin/dbus-daemon" sauid=106 hostname=? addr=? terminal=?'
+Jul 18 21:35:49 XXXXXXXX kernel: [ 5027.856093] audit: type=1107 audit(1468895749.317:2881): pid=886 uid=106 auid=4294967295 ses=4294967295 msg='apparmor="ALLOWED" operation="dbus_method_call"  bus="system" path="/org/freedesktop/login1" interface="org.freedesktop.login1.Manager" member="CanPowerOff" mask="send" name="org.freedesktop.login1" pid=10058 label="snap.kodi.kodi" peer_pid=903 peer_label="unconfined"
+Jul 18 21:35:49 XXXXXXXX kernel: [ 5027.856093]  exe="/usr/bin/dbus-daemon" sauid=106 hostname=? addr=? terminal=?'
+Jul 18 21:35:49 XXXXXXXX kernel: [ 5027.860905] audit: type=1107 audit(1468895749.325:2882): pid=886 uid=106 auid=4294967295 ses=4294967295 msg='apparmor="ALLOWED" operation="dbus_method_call"  bus="system" path="/org/freedesktop/login1" interface="org.freedesktop.login1.Manager" member="CanReboot" mask="send" name="org.freedesktop.login1" pid=10058 label="snap.kodi.kodi" peer_pid=903 peer_label="unconfined"
+Jul 18 21:35:49 XXXXXXXX kernel: [ 5027.860905]  exe="/usr/bin/dbus-daemon" sauid=106 hostname=? addr=? terminal=?'
+
```
Comment (dholbach) [inline kodi-stable/README.md]: Do you know if they were discussed on snapcraft@lists.snapcraft.io? It looks like we might need another interface or two.

File: kodi-stable/setup/gui/kodi.desktop [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Comment=Manage and view your media
+Exec=kodi
+Icon=${SNAP}/meta/gui/kodi.png
+Terminal=false
+Type=Application
+Categories=AudioVideo;Video;Player;TV;
+
+Actions=Fullscreen;Standalone;
+
+[Desktop Action Fullscreen]
```
Comment (dholbach) [inline kodi-stable/setup/gui/kodi.desktop]: Do these actions work as expected?

**Comment by dholbach** [comment]
In Travis this fails to build with ```
checking mpeg2dec/mpeg2.h usability... no

checking mpeg2dec/mpeg2.h presence... no

checking for mpeg2dec/mpeg2.h... no

configure: error: Could not find a required library. Please see the README for your platform.

Command '['/bin/sh', '/tmp/tmpudwgxd8_', './configure', '--prefix=', '--prefix=/']' returned non-zero exit status 1
```

**Comment by dholbach** [comment]
On my machine locally it fails with
```
checking for LIBDCADEC... no
configure: error: libdcadec not found
Command '['/bin/sh', '/tmp/tmpwwmf3jxr', './configure', '--prefix=', '--prefix=/']' returned non-zero exit status 1
```

File: kodi-stable/README.md (hunk 1 above)
Comment (dgmvecuador) [inline kodi-stable/README.md]: I dont think so, maybe is a new issue and I havent reported it

File: kodi-stable/setup/gui/kodi.desktop (hunk 2 above)
Comment (dgmvecuador) [inline kodi-stable/setup/gui/kodi.desktop]: Yes, they work as expected

kodi_desktop_snap

File: kodi-stable/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      - camera
+
+parts:
+  kodi:
+    source: https://github.com/xbmc/xbmc.git
+    source-tag: 16.1-Jarvis
+    plugin: x_kodiautotools
+    configflags:
+      - --prefix=/
+    build-packages:
```
Comment (dholbach) [inline kodi-stable/snapcraft.yaml]: Can you apply this? http://paste.ubuntu.com/21252024/

It add's a few `build-packages`.

**Comment by dholbach** [comment]
Can we somehow adjust where ffmpeg comes from? This is how the build fails (both using 'snapcraft' or 'snapcraft cleanbuild'): http://paste.ubuntu.com/21252125/

File: kodi-stable/README.md (hunk 1 above)
Comment (dholbach) [inline kodi-stable/README.md]: I think it'd be good to at least bring it up there and see if others can work on resolving them.

File: kodi-stable/setup/gui/kodi.desktop (hunk 2 above)
Comment (dholbach) [inline kodi-stable/setup/gui/kodi.desktop]: :+1:<a href='#crh-update-complete'></a>

**Comment by dholbach** [comment]
:+1:<a href='#crh-update-none'></a>

**Comment by dgmvecuador** [comment]
Kodi developers usually recommend to use version of ffmpeg provided inside the code, so we should use that one.  They update ffmpeg each major release.

**Comment by dholbach** [comment]
Ok, let's pick whatever works. Can you apply the patch I specified earlier? That way you should see in the next Travis build how it's currently failing. Currently it's missing some build-packages.

**Comment by dgmvecuador** [comment]
I have applied your patch and other changes. Now the build is successful in Travis

**Comment by dholbach** [comment]
Great work!

When I try to run it, I see this though:
daniel@daydream:~/dev/snappy/snappy-playpen/kodi-stable$ kodi
libGL error: unable to load driver: i965_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: i965
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast

And I get a black screen. Anything I'm missing?

**Comment by dgmvecuador** [comment]
Its a problem with opengl libs, there are multiple bugs reported like https://bugs.launchpad.net/snappy/+bug/1574851

As workaround, you can copy your driver libs to the folder "/var/lib/snapd/lib/gl/" in my case, I copied all the content of "/usr/lib/nvidia-xxx". I dont know about your graphic card, but it seems to work with intel without problems.

**Comment by dholbach** [comment]
I'm on intel too (i915).

**Comment by dgmvecuador** [comment]
You should copy all the libs from Intel, "i965_dri.so" and the dependencies to "/var/lib/snapd/lib/gl/", maybe check the deb package from the repository.

**Comment by dgmvecuador** [comment]
Hi, did you solve the problem in your environment??

**Comment by dholbach** [comment]
Sorry for the long wait. This makes it work for me: http://paste.ubuntu.com/22795846/

**Comment by dholbach** [comment]
Brilliant work! :-D

**Pull request merged**